makefile.conf.in

```diff
@@ -65,8 +65,6 @@ ifeq ($(ARCH),LINUX)
   ifndef AUDIODRIVER
 	AUDIODRIVER=OSS
   endif
-  CC = gcc
```

Comment:
I need a bit more annotation about changes like this.  Why the change, and where do these get set now?

Reply:
The general answer (for this and the other changes) is that it's whatever I found necessary get this to build with [Emscripten](https://emscripten.org/).

The reason for this particular change is that Emscripten is based on LLVM, not gcc, and furthermore uses its own wrapper for clang (emcc). And in any case I didn't see a reason to insist on gcc on Linux; the user might prefer clang for standalone builds.

Unsetting `CC` allows an externally-provided `CC` to take precedence. Emscripten sets this to the path to `emcc` (via `emconfigure` and `emmake` as in apps/wasmtest/README; see [here](https://emscripten.org/docs/compiling/Building-Projects.html) and [here](https://github.com/emscripten-core/emscripten/blob/main/tools/building.py) for more information). If it's _not_ set externally, it defaults to `cc`, which seems like a reasonable default on Linux.

We could use `?=` if we want to specify `gcc` as the default while still allowing an external `CC` to override it. But I'm not sure why all of these variables need to be manually specified for each architecture in the first place; my impression is that `./configure` is supposed to determine sane values for `CC`, `LD`, etc. for "normal" builds.